resource/interface_factories.cpp.em

```diff
@@ -277,7 +277,15 @@ void ServiceFactory<
 ) {
 @[      for field in service["fields"][type.lower()]]@
 @[        if field["array"]]@
+@[          if field["is_upper_bound"]]@
+  // ensure array size
+@[            if field["array_size"] > 0 and frm == "1"]@
+  // check boundary
+                assert(req@(frm).@(field["ros1"]["name"]).size() <= @(field["array_size"]));
```

Comment:
Based on the other comments I don't think this `assert` is ever being generated.